modules/ROOT/pages/_partials/sync/js-client-realtime-updates.adoc

```diff
@@ -1,3 +1,77 @@
 = Realtime Updates
 
-TODO
\ No newline at end of file
+Subscriptions are a core concept of GraphQL and one which allow us to enable real time updates of data
+from the server to the client.
+To do this GraphQL Subscriptions utilise websockets to enable clients to subscribe to published changes.
+
+The architecture of websockets is as follows:
+
+* Client connects to websocket server
+* Upon certain events, the server can publish the results of these events to the websocket
+* Any _currently connected_ client to that websocket will receive these results
+* The client can close the connection at any time and will no longer recieve updates
+
+Websockets are not however, a queueing system. For example if the client is offline when the server publish the changes
+the client will not receive them. To receive updates the client must be currently connected to the websocket server,
+updates made over this websocket while the client is offline will not be replayed.
+{sync-client} comes with subscription support out of the box including auto-reconnection upon device restart or network reconnect.
+To enable subscriptions on your client set the following
+paramater in the {sync-client} config object. A DataSyncConfig interface is also available from {sync-client} if you wish to use it.
+
+== Setting up a Client to use Subscriptions
+
+To set up a client to use subscriptions simply provide a wsUrl in your `mobile-services.json` file, {sync-client} will automatically
+apply the necessary logic to handle websocket requests. This can also be provided manually as seen below.
+
+```
+const config: DataSyncConfig {
+    ...
+    wsUrl: { WEB_SOCKET_URL_HERE }
+    ...
+}
+```
+Where wsUrl is the full URL of the websocket endpoint of your GraphQL server. Once this object is created, use it to initialise {sync-client} with:
+
+```
+import { createClient } from '@aerogear/voyager-client';
+
+const client = createClient(config)
+```
+
+== Using Subscriptions
+
+Once the client is configured we can then make use of subscriptions as follows.
+
+```
+
+getTasks() {
+  const tasks = client.watchQuery({
+    query: GET_TASKS
+  });
+
+  tasks.subscribeToMore({
+    document: TASK_ADDED_SUBSCRIPTION,
+    updateQuery: (prev, { subscriptionData }) => {
+      if(subscribtionData.data){
+        const newTask = subscribtionData.data.taskAdded;
+        return allTasks: [
+            ...prev,
+            newItem
+          ];
+        } else {
+          return prev
+        }
+    }
+  });
+  return tasks;
+}
+```
+
+This `getTasks()` function can then be used from a client service to control and view the updated tasks recieved
+from the server. An example can be seen below.
+
+```
+getTasks().subscribe(result => {
+  this.items = result.data.allTasks
+}
+```
```

Comment:
are we missing some content on the client? For example, is there anything developers need to do when the app is online from the offline state? We said at the beginning that the client will miss all the subscription messages when it's offline, so when it's online again, should user run some queries again to get the data they probably have missed?

Reply:
Another question I think worth a bit more explaining is the cache & subscription. Is it handled automatically by the voyager server, or do developers need to handle that?

Reply:
+1

Reply:
@wei-lee to your first point, I can state that we have a network interface available in the sdk and that the developer can use that to detect network changes and therefore run some queries. I'm not sure how deep into that implementation I should go though - a lot of it will be very app specific. Wrt the cache - the developer will need to do some deduplication. We have an example of how we handle this but I am hesitant to put that code in the docs as again, its probably implementation specific. What do you guys think?